app/javascript/shared/components/focusTrap.jsx

```diff
@@ -1,27 +1,62 @@
 import PropTypes from 'prop-types';
+import { h, Fragment } from 'preact';
 import { useLayoutEffect, useRef } from 'preact/hooks';
 import { createFocusTrap } from 'focus-trap';
 import { defaultChildrenPropTypes } from '../../common-prop-types';
+import { KeyboardShortcuts } from './useKeyboardShortcuts';
 
-export const FocusTrap = ({ selector, children }) => {
+export const FocusTrap = ({ selector, children, onDeactivate }) => {
   const focusTrap = useRef(null);
 
+  const currentLocationHref = document.location.href;
+
+  const routeChangeObserver = new MutationObserver((mutations) => {
```

Comment:
There was an issue with the article "unsaved changes" modal where the trap wasn't being deactivated on navigation via the confirm url. This just makes sure any current trap is always deactivated on navigation away